### ISSUE TITLE
Invalid behaviors in assignment/comparison

### ISSUE THREAD

**Issue by hachi8833** [goby-lang/goby#260]
Checked with REPL for now.

* On invalid assignments, Ruby returns 'unexpected '=', expecting end-of-input' correctly, but Goby returns the right side term:

```ruby
"v" = "a"       #=> a
"v" = 1         #=> 1
{ a: 1 } = 1    #=> 1
[1, 2] = 1      #=> 1
```

* Comparisons between different types does not work correctly:

```ruby
"v" == "a"         #=> false   -- works
"v" == 1           #=> TypeError: Expect argument to be String. got=Integer
1 == "v"           #=> expect argument to be Integer type
1 != [1,2,3]       #=> expect argument to be Integer type
[1,2,3] != 1       #=> UndefinedMethodError: Undefined Method '!=' for [1, 2, 3]
[1, 2, 3] == [1]   #=> Undefined Method '!=' for [1, 2, 3]
[1, 2, 3] == [1]   #=> UndefinedMethodError: Undefined Method '==' for [1, 2, 3] 
```
In addition, the error messages are inconsistent.

They should be fixed, as well as adding tests for all combination of objects.

**Comment by st0012** [comment]
The error styling issue is related to #237 and I'll fix the invalid assignment issue.

**Comment by hachi8833** [comment]
Thank you very much!
I'm checking all possible combinations between Ruby and Goby, and will report here.

**Comment by Maxwell-Alexius** [comment]
I think I know how to solve the comparison part. Working on it.

**Comment by st0012** [comment]
This seems to be fixed in #285
